### PR TITLE
Fix formatting when making an accessor readonly

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/69876

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs [hunk 1]
```diff
@@ -40,7 +40,6 @@ class {|FixAllInContainingType:Program|}
 
         var expected = """
             Console.WriteLine("Hello, World!");
-
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersFixAllTests.cs:43]: this result is better.  the final formatting matches the initial formatting.

File: src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs [hunk 2]
```diff
@@ -114,7 +114,6 @@ public event Action e5 { add { } remove { } }
                         event Action I.e6 { add { } remove { } }
 
                         static C() { }
-
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs:117]: this result is better.  the final formatting matches the initial formatting.

File: src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs [hunk 3]
```diff
@@ -213,36 +213,37 @@ static async Task<int> fibonacci(int n)
...
-        }
-    }
-}
-""",
-parseOptions: CSharp8ParseOptions);
+            class C
+            {
+                void M()
+                {
+                    int x;{{leadingTrivia}}
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:238]: this result is better.  the final formatting matches the initial formatting.

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:238]: same for the rest of the results.

File: src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs [hunk 4]
```diff
@@ -343,15 +345,14 @@ void M()
...
-"""
+$$"""
 using System;
 
 class C
 {
     void M()
     {
-        bool otherFunction() => true;
-
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:354]: this result is better.  the final formatting matches the initial formatting.

File: src/Analyzers/CSharp/Tests/MakeMemberStatic/MakeMemberStaticTests.cs [hunk 5]
```diff
@@ -133,11 +133,8 @@ namespace NS
                 public static class Foo
                 {
                     static int i;
-
                     static void M() { }
-
                     static object P { get; set; }
-
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/MakeMemberStatic/MakeMemberStaticTests.cs:140]: this result is better.  the final formatting matches the initial formatting.

File: src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs [hunk 6]
```diff
@@ -53,7 +53,6 @@ static void Main(string[] args)
 
         var expected = """
             using System;
-
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs:56]: this result is better.  the final formatting matches the initial formatting.

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 7]
```diff
@@ -1741,6 +1750,17 @@ private static SyntaxTokenList AsModifierList(Accessibility accessibility, Decla
         if (modifiers.IsPartial)
             list.Add(PartialKeyword);
 
+        for (int i = 0, n = list.Count; i < n; i++)
+        {
+            // By default, do not place leading elastic trivia on modifiers we make.  Just because something is
+            // adding/removing/modifying modifiers does not mean we want the parent construct to change its formatting
+            // respective to what's around it.
+            if (!withLeadingElasticMarker)
+                list[i] = list[i].WithoutLeadingTrivia();
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:1759]: the crux of the fix.  we do *not* want elastic trivia at teh start of a member by default, and especially not if we're just mutating modifiers.  We want the modifiers to just be inertly placed into the member, leaving the formatting the same as how it started.  

callers can choose if they do want leading elastic trivia added

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**